dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
 }
 
-// Transformations
+func (ctx *CanvasRenderingContext2D) Rect(x, y, width, height float64) {
+	ctx.Call("rect", x, y, width, height)
+}
 
-func (ctx *CanvasRenderingContext2D) Scale(scaleWidth, scaleHeight int) {
-	ctx.Call("scale", scaleWidth, scaleHeight)
+//Drawing paths
+
+func (ctx *CanvasRenderingContext2D) Fill() {
+	ctx.Call("fill")
+}
+
+func (ctx *CanvasRenderingContext2D) Stroke() {
+	ctx.Call("stroke")
+}
+
+func (ctx *CanvasRenderingContext2D) DrawFocusIfNeeded(element HTMLElement, path *js.Object) {
+	ctx.Call("drawFocusIfNeeded", element, path)
 }
 
+func (ctx *CanvasRenderingContext2D) ScrollPathIntoView(path *js.Object) {
+	ctx.Call("scrollPathIntoView", path)
+}
+
+func (ctx *CanvasRenderingContext2D) Clip() {
+	ctx.Call("clip")
+}
+
+func (ctx *CanvasRenderingContext2D) IsPointInPath(x, y float64) bool {
+	return ctx.Call("isPointInPath", x, y).Bool()
+}
+
+func (ctx *CanvasRenderingContext2D) IsPointInStroke(path *js.Object, x, y float64) bool {
+	return ctx.Call("isPointInStroke", path, x, y).Bool()
+}
+
+// Transformations
+
 func (ctx *CanvasRenderingContext2D) Rotate(angle float64) {
 	ctx.Call("rotate", angle)
 }
 
-func (ctx *CanvasRenderingContext2D) Translate(x, y int) {
+func (ctx *CanvasRenderingContext2D) Scale(scaleWidth, scaleHeight float64) {
+	ctx.Call("scale", scaleWidth, scaleHeight)
+}
+
+func (ctx *CanvasRenderingContext2D) Translate(x, y float64) {
 	ctx.Call("translate", x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) Transform(a, b, c, d, e, f int) {
+func (ctx *CanvasRenderingContext2D) Transform(a, b, c, d, e, f float64) {
 	ctx.Call("transform", a, b, c, d, e, f)
 }
 
-func (ctx *CanvasRenderingContext2D) SetTransform(a, b, c, d, e, f int) {
+func (ctx *CanvasRenderingContext2D) SetTransform(a, b, c, d, e, f float64) {
 	ctx.Call("setTransform", a, b, c, d, e, f)
 }
 
-// Text
+//resetTransform
 
-func (ctx *CanvasRenderingContext2D) FillText(text string, x, y, maxWidth int) {
-	if maxWidth == -1 {
-		ctx.Call("fillText", text, x, y)
-		return
-	}
+// Drawing images
 
-	ctx.Call("fillText", text, x, y, maxWidth)
+func (ctx *CanvasRenderingContext2D) DrawImage(image *Element, dx, dy float64) {
+	ctx.Call("drawImage", image, dx, dy)
 }
 
-func (ctx *CanvasRenderingContext2D) StrokeText(text string, x, y, maxWidth int) {
-	if maxWidth == -1 {
-		ctx.Call("strokeText", text, x, y)
-		return
-	}
+func (ctx *CanvasRenderingContext2D) DrawImageWithDst(image *Element, dx, dy, dWidth, dHeight float64) {
+	ctx.Call("drawImage", image, dx, dy, dWidth, dHeight)
+}
 
-	ctx.Call("strokeText", text, x, y, maxWidth)
+func (ctx *CanvasRenderingContext2D) DrawImageWithSrcAndDst(image *Element, sx, sy, sWidth, sHeight, dx, dy, dWidth, dHeight float64) {
+	ctx.Call("drawImage", image, sx, sy, sWidth, sHeight, dx, dy, dWidth, dHeight)
+}
+
+// Pixel manipulation
+
+func (ctx *CanvasRenderingContext2D) CreateImageData(width, height float64) *ImageData {
```

Comment:
I think we should use `int`s for indexes and dimensions. ImageData represents pixel data, and pixels ("picture elements") are distinct elements. As far as I can tell, there can't be valid floating point values for width and height, etc.

![image](https://user-images.githubusercontent.com/1924134/29585460-2da5155c-8755-11e7-820c-017812af0b51.png)

Reply:
This isn't done yet.

Reply:
Changed to int.